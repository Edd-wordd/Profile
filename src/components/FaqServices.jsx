import React from 'react'
import SectionHeader from '../components/SectionHeader'
import Container from '@material-ui/core/Container'
import { makeStyles, Grid, Typography } from '@material-ui/core'
import Accordion from '@material-ui/core/Accordion'
import AccordionSummary from '@material-ui/core/AccordionSummary'
import AccordionDetails from '@material-ui/core/AccordionDetails'
import ComputerIcon from '@material-ui/icons/Computer'
import mainTheme from '../theme'
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';


const useStyles = makeStyles((theme) => ({
    root: {
     
    },
    accordion:{
        width: '600px',
        backgroundColor: mainTheme.palette.defaultLight.main,
    },
    mainInfo: {
        padding: '2rem 0rem'
    },
    questionWrapper: {
        padding: '1rem 0rem 3rem 0rem'
    }
   
  }))

const questions =
[
    {
        question: "this is one question",
        details: 'this is the answer to the question'
    }, {
        question: "this is Two question",
        details: 'this is the answer to the question'
    }, {
        question: "this is Three question",
        details: 'this is the answer to the question'
    }, {
        question: "this is Four question",
        details: 'this is the answer to the question'
    }, {
        question: "this is Five question",
        details: 'this is the answer to the question'
    }, 

]
const questionSecond = [
    {
        question: "this is Six question",
        details: 'this is the answer to the question'
    }, {
        question: "this is Seven question",
        details: 'this is the answer to the question'
    }, {
        question: "this is Eight question",
        details: 'this is the answer to the question'
    }, {
        question: "this is Nine question",
        details: 'this is the answer to the question'
    }, {
        question: "this is Ten question",
        details: 'this is the answer to the question'
    }, 
]
    

function FaqServices() {
    const classes = useStyles()
    const [expanded, setExpanded] = React.useState('panel1')

    const handleChange = (panel) => (event, isExpanded) => {
        setExpanded(isExpanded ? panel: false)
    }
    return (
        <div>
            <SectionHeader  title="FAQ" subTitle="Ask Away"/>
            <Container maxWidth='lg'>
                <Grid container direction='row' justifyContent='center' alignItems='center' className={classes.mainInfo}>
                <Typography paragraph >
                    Lorem ipsum dolor sit amet consectetur adipisicing elit. Amet veniam ipsum aliquid explicabo. 
                    Harum ipsa blanditiis eaque ducimus dolores libero corrupti eligendi debitis. Nulla delectus tempore aliquid. 
                    Beatae, blanditiis. Necessitatibus? Lorem ipsum dolor sit amet consectetur, adipisicing elit.
                    Nemo quasi beatae, doloremque aliquam deserunt veritatis perspiciatis molestias sequi libero dolorem, natus quibusdam ex atque, sit sapiente odit quidem eum autem.
                </Typography>
                </Grid>
                <Grid container direction='row' justify='space-evenly' alignItems='center' className={classes.questionWrapper}>
                    <Grid>
                        {questions.map((question, index) => {
                            return(
                    <Accordion className={classes.accordion} key={index} expanded={expanded === index} onChange={handleChange(index)}>
                        <AccordionSummary  expandIcon={<ExpandMoreIcon/>}>
                            <Typography>{question.question}</Typography>
                        </AccordionSummary>
                        <AccordionDetails>
                            <Typography>
                                This is in the details
                            </Typography>
                        </AccordionDetails>
                    
                </Accordion>
               
                        )})}
                        </Grid>
                        <Grid>
                        {questionSecond.map((question, index) => {
                            return(
                    <Accordion className={classes.accordion} key={index}expanded={expanded === index} onChange={handleChange(index)}>
                        <AccordionSummary  expandIcon={<ExpandMoreIcon/>}>
                            <Typography>{question.question}</Typography>
                        </AccordionSummary>
                        <AccordionDetails>
                            <Typography>
                                This is in the details
                            </Typography>
                        </AccordionDetails>
                    
                </Accordion>
               
                        )})}
                        </Grid>
           
                </Grid>
            </Container>
            
        </div>
    )
}

export default FaqServices
