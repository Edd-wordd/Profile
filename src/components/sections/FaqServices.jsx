import React from 'react'
import SectionHeader from '../../components/sections/SectionHeader'
import { Grid, Typography, Container } from '@mui/material'
import { Accordion, AccordionSummary, AccordionDetails } from '@mui/material'
import ExpandMoreIcon from '@mui/icons-material/ExpandMore'
import { CustomAccordion } from '../styles/sections/FaqServices.styles'
import faqQuestions from '../../data/faqData'

function FaqServices() {
  const [expanded, setExpanded] = React.useState('panel1')

  const handleChange = (panel) => (event, isExpanded) => {
    setExpanded(isExpanded ? panel : false)
  }
  return (
    <>
      <SectionHeader title="FAQ" subTitle="Ask Away" />
      <Container maxWidth="lg" sx={{ paddingBottom: '4rem' }}>
        <Grid sx={{ padding: '3rem 0rem' }}>
          <Typography component="p">
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Amet veniam ipsum aliquid
            explicabo. Harum ipsa blanditiis eaque ducimus dolores libero corrupti eligendi debitis.
            Nulla delectus tempore aliquid. Beatae, blanditiis. Necessitatibus? Lorem ipsum dolor
            sit amet consectetur, adipisicing elit. Nemo quasi beatae, doloremque aliquam deserunt
            veritatis perspiciatis molestias sequi libero dolorem, natus quibusdam ex atque, sit
            sapiente odit quidem eum autem.{' '}
          </Typography>
        </Grid>
        <Grid container justifyContent="center">
          <Grid item xs={12} sm={8} md={8} lg={8} xl={8}>
            <Grid component="section" aria-labelledby="faq-section">
              {faqQuestions.map((question) => {
                return (
                  <CustomAccordion key={question.id}>
                    <Accordion
                      key={question.id}
                      expanded={expanded === `${question.rank}`}
                      onChange={handleChange(`${question.rank}`)}
                    >
                      <AccordionSummary expandIcon={<ExpandMoreIcon />}>
                        <Typography component="h2">{question.questions}</Typography>
                      </AccordionSummary>
                      <AccordionDetails>
                        <Typography component="p">{question.answers}</Typography>
                      </AccordionDetails>
                    </Accordion>
                  </CustomAccordion>
                )
              })}
            </Grid>
          </Grid>
        </Grid>
      </Container>
    </>
  )
}

export default FaqServices
