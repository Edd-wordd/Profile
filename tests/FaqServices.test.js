import React from 'react'
import testComponent from './testComponent'
import FaqServices from '../src/components/sections/FaqServices'

testComponent(<FaqServices />, 'FaqServices')
