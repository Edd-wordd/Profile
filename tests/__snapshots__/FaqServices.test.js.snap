// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FaqServices should render FaqServices 1`] = `"Component"`;
